lib/ansible/modules/windows/win_wakeonlan.ps1

```diff
@@ -0,0 +1,63 @@
+#!powershell
+# This file is part of Ansible
+#
+# (c) 2017, Dag Wieers <dag@wieers.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$mac = Get-AnsibleParam -obj $params -name "mac" -type "str"
+$broadcast = Get-AnsibleParam -obj $params -name "broadcast" -type "str" -default "255.255.255.255"
+$port = Get-AnsibleParam -obj $params -name "port" -type "int" -default 7
+
+$result = @{
+    changed = $true
+}
+
+Function Wake-On-Lan($mac, $broadcast, $port, $whatif) {
+    $mac_orig = $mac
+    $broadcast = [Net.IPAddress]::Parse($broadcast)
+
+    # Remove possible separator from MAC address
+    if ($mac.Length -eq (12 + 5)) {
+        $mac = $mac.Replace($mac.Substring(2, 1), "")
+    }
+
+    # If we don't end up with 12 hexadecimal characters, fail
+    if ($mac.Length -ne 12) {
+        Fail-Json @{} "Incorrect MAC address length: $mac_orig"
```

Comment:
Well, within the function $result was not exposed, but since we don't have a function, this is no longer an issue.